src/turret.cpp

```diff
@@ -317,7 +317,7 @@ int turret_data::fire( Character &c, const tripoint &target )
     gun_mode mode = base()->gun_current_mode();
 
     prepare_fire( c );
-    shots = c.fire_gun( target, mode.qty, *mode );
+    shots = c.fire_gun( target, mode.qty, *mode, item_location() );
```

Comment:
Why is this necessary? You have supplied a default argument in the prototype of character::fire_gun()

Reply:
Just followed the way they did in the BN PR, will remove this.